unittests/DOSGuard.cpp

```diff
@@ -59,7 +64,7 @@ constexpr static auto IP = "127.0.0.2";
 class FakeSweepHandler
 {
 private:
-    using guard_type = BasicDOSGuard<FakeSweepHandler>;
+    using guard_type = BasicDOSGuard<WhitelistHandler, FakeSweepHandler>;
```

Comment:
This test is for BasicDOSGuard not for the whitelist. Let's create a `Fake` or `Mock` for the `WhitelistHandler` and pass it in here. You can then just check that if your `isWhiteListed()` function of DOSguard returns the same thing as the mock/fake and that's enough here.

For the WhitelistHandler you would have a separate file with a separate test suite 🥇